### PR TITLE
docs(constants): document missing constants

### PR DESCRIPTION
**Pull request by almeidx**
**Please describe the changes this PR makes and why it should be merged:**
- Several keys were missing from the `Constants` typedef
- `InteractionTypes`, `InteractionResponseTypes`, and `MessageComponentTypes` were missing the modal related types.
- `SweeperKeys` was missing in the typings
- Sorted the `Constants` typedef alphabetically 
- Backported #8179

**Status and versioning classification:**
- I know how to update typings and have done so, or typings don't need updating
- This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/util/Constants.js [hunk 1]
```diff
@@ -127,6 +158,85 @@ exports.Opcodes = {
...
+ * * STAGE_INSTANCE_DELETE
+ * * GUILD_STICKER_CREATE
+ * * GUILD_STICKER_DELETE
+ * * GUILD_STICKER_UPDATE
+ * * GUILD_SCHEDULED_EVENT_CREATE
+ * * GUILD_SCHEDULED_EVENT_UPDATE
+ * * GUILD_SCHEDULED_EVENT_DELETE
+ * * GUILD_SCHEDULED_EVENT_USER_ADD
+ * * GUILD_SCHEDULED_EVENT_USER_REMOVE
+ * @typedef {string} Event
```
Comment (vladfrangu) [inline src/util/Constants.js:238]: Thats not the type

File: src/util/Constants.js [hunk 2]
```diff
@@ -214,6 +324,16 @@ exports.Events = {
...
 
+/**
+ * The types of events emitted by a Shard:
+ * * CLOSE
+ * * DESTROYED
+ * * INVALID_SESSION
+ * * READY
+ * * RESUMED
+ * * ALL_READY
+ * @typedef {string} ShardEvent
```
Comment (vladfrangu) [inline src/util/Constants.js:335]: Ditto

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: src/util/Constants.js [hunk 3]
```diff
@@ -127,23 +158,93 @@ exports.Opcodes = {
...
-   * @deprecated See {@link https://github.com/discord/discord-api-docs/issues/3690 this issue} for more information.
-   */
   APPLICATION_COMMAND_CREATE: 'applicationCommandCreate',
-  /**
-   * @deprecated See {@link https://github.com/discord/discord-api-docs/issues/3690 this issue} for more information.
-   */
   APPLICATION_COMMAND_DELETE: 'applicationCommandDelete',
-  /**
-   * @deprecated See {@link https://github.com/discord/discord-api-docs/issues/3690 this issue} for more information.
-   */
```
Comment (almeidx) [inline src/util/Constants.js:146]: These were crashing the docgen

**Review by vladfrangu — APPROVED** [review]

File: src/util/Constants.js (hunk 3 above)
Comment (iCrawl) [reply src/util/Constants.js:146]: How exactly? 🤔

Comment (ImRodry) [reply src/util/Constants.js:146]: Seems to be this run https://github.com/discordjs/discord.js/runs/7042040198?check_suite_focus=true

Comment (almeidx) [reply src/util/Constants.js:146]: Yep, the "Tests / Documentation (pull_request)" check in the first commit

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**